.gitignore

```diff
@@ -63,6 +63,8 @@ nbproject/
 .idea
 cmake-build-*/*
 coverage-report/
+.direnv
+result
```

Comment:
Please add a comment describing what `result` is.

```suggestion
# nix build output symlink
# described at https://nix.dev/manual/nix/2.18/command-ref/nix-build.html?highlight=result#examples 
result/
```

Not a perfect link but imo nix documentation is lacking. The link could be omitted.